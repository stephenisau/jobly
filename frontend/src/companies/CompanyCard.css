.company-logo {
    width: 200px;
    height: 200px;
    float: left;
    margin: auto;
}

.logo {
    justify-items: center;
}

.Card {
    border: 1px solid;
    padding: 10px;
    box-shadow: 5px 10px 18px black;
    min-height: 250px;
}

.card > a {
    color: black;
}
.card-description {
    bottom: 3px;
    right: 3px
}

.company-description {
    text-align: center;
}