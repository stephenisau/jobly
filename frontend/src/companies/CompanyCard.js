import React, { Component } from "react";
import { Link } from "react-router-dom";

class CompanyCard extends Component {
  render() {
    let img = this.props.company.logo_url ?
      <img src="https://png.pngtree.com/svg/20170307/c02483909c.svg" alt={`${this.props.company.name} logo`} /> : <img src={this.props.company.logo_url} alt={`${this.props.company.name} logo`} />
    return (
      <Link exact to={`companies/${this.props.company.handle}`}>
        <div className="Card card mt-3">
          <div className="card-body">
            <h6>{this.props.company.name}</h6>
            <p>{this.props.company.description}</p>
            {img}
          </div>
        </div>
      </Link>

    );
  }
}

export default CompanyCard;