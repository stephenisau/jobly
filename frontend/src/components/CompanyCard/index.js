import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./CompanyCard.css";


/*
Display component deailing with showing company information.
*/

class CompanyCard extends Component {
  render() {
    const imageSrc = this.props.company.logo_url ? this.props.company.logo_url : "https://png.pngtree.com/svg/20170307/c02483909c.svg"
    return (
      <div className="card mt-2 Card">
        <Link to={`companies/${this.props.company.handle}`}>
          <div className="card-body">
            <div className="logo">
              <img className="company-logo" src={imageSrc} alt="Company" />
            </div>
            <div className="company-description">
              <h3 className="card-title text-center">{this.props.company.name}</h3>
              <p className="justify-content-end mb-2 card-description">{this.props.company.description}</p>
            </div>
          </div>
        </Link>
      </div>

    );
  }
}

export default CompanyCard;