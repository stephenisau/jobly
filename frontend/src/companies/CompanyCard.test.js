import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CompanyCard from './CompanyCard';
import { BrowserRouter as Router } from 'react-router-dom';



describe("Company Card", () => {
  it("renders company card", () => {
    const company = {
      logo_url: "https://png.pngtree.com/svg/20170307/c02483909c.svg",
      name: "testCompany",
      description: "This is a test company",
      handle: "random"
    }
    const tree = renderer.create(
      <Router>
        <CompanyCard company={company} />
      </Router>
    ).toJSON()
    expect(tree).toMatchSnapshot();
  });
})

