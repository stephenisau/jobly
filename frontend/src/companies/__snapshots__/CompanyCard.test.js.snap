// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company Card renders company card 1`] = `
<div
  className="card mt-2 Card"
>
  <a
    href="/companies/random"
    onClick={[Function]}
  >
    <div
      className="card-body"
    >
      <div
        className="logo"
      >
        <img
          alt="Company"
          className="company-logo"
          src="https://png.pngtree.com/svg/20170307/c02483909c.svg"
        />
      </div>
      <div
        className="company-description"
      >
        <h3
          className="card-title text-center"
        >
          testCompany
        </h3>
        <p
          className="justify-content-end mb-2 card-description"
        >
          This is a test company
        </p>
      </div>
    </div>
  </a>
</div>
`;
